druid-handler/src/java/org/apache/hadoop/hive/druid/serde/DruidWritable.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.druid.serde;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.io.Writable;
+
+import com.google.common.base.Objects;
+
+/**
+ * Writable for Druid results.
+ */
+public class DruidWritable implements Writable {
+
+  private final Map<String, Object> value;
```

Comment:
@ashutoshc, using array instead of map does not seem straightforward.

The map, except for GroupBy queries, is taken directly from the result of the Druid query as it is. The record reader just pulls the results from Druid, but it does not have additional information about the schema of the results e.g. order of the columns needed.

I think with some re-engineering of the code, I could pass the column names to the RecordReader. However, that would just mean moving the work from the deserializer to the reader, as we would still need a look up in the Map in order to extract the columns and insert them in the array of the Writable object. Thus, two look-ups are needed: one at the reader, and one at the deserializer.

I think the only gain would be that we do not store column names in every record. However (correct me if I am wrong) the deserializer is called after the record is produced by the reader, thus there is no network transfer? And those objects are already in memory? Hence, I am not sure if we should change it. What do you think?